From 94dc9287caa16ee0aa6e8faa73b2d6d9707d1f46 Mon Sep 17 00:00:00 2001
From: Antheas Kapenekakis <lkml@antheas.dev>
Date: Sun, 2 Nov 2025 16:58:58 +0100
Subject: [PATCH 108/112] platform/x86: asus-wmi: Don't reset charge threshold
 on boot

Currently, the driver resets the charge threshold to 100% on every boot.
This is because the driver cannot read the current threshold from the
EC. The EC is perfectly capable of storing it across reboots.

This causes e.g., Powerdevil, to think the charge limit is 100% on boot
and for users to have to set it every time. Just skip setting it on boot
even if this causes userspace to drift.

Fixes: 7973353e92ee ("platform/x86: asus-wmi: Refactor charge threshold to use the battery hooking API")
Signed-off-by: Antheas Kapenekakis <lkml@antheas.dev>
---
 drivers/platform/x86/asus-wmi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/platform/x86/asus-wmi.c b/drivers/platform/x86/asus-wmi.c
index 1046ad699..5f2389475 100644
--- a/drivers/platform/x86/asus-wmi.c
+++ b/drivers/platform/x86/asus-wmi.c
@@ -1541,7 +1541,7 @@ static int asus_wmi_battery_add(struct power_supply *battery, struct acpi_batter
 	 * and we can't get the current threshold so let set it to 100% when
 	 * a battery is added.
 	 */
-	asus_wmi_set_devstate(ASUS_WMI_DEVID_RSOC, 100, NULL);
+	// asus_wmi_set_devstate(ASUS_WMI_DEVID_RSOC, 100, NULL);
 	charge_end_threshold = 100;
 
 	return 0;
-- 
2.52.0

